src/neurostatslib/basis.py

```diff
@@ -465,6 +462,13 @@ def _generate_knots(
         )
         return self.knot_locs
 
+    @staticmethod
+    def _check_samples_non_empty(sample_pts):
+        if sample_pts.shape[0] == 0:
+            raise ValueError(
+                "Empty sample array provided. At least one sample is required for evaluation!"
```

Comment:
why is this here and only for BSpline?

from the tests, looks like you're expecting it when `evaluate_on_grid(x)` for `x<=0`, but is that really an issue? for the negative case, we'll get an error in `np.linspace` and for `x=0`, you'd get an empty array otherwise, which seems reasonable. Is there a specific reason to check for this?

and in general, don't like that we'll have different behavior for BSpline than for the rest, when really, it's the same concerns, right?

Reply:
If you don't check, with x=0 you get an "invalid input data" error message from the scipy splev call. 

I am fine with that but:
- it is a bit nested and less clear for a user what may have caused it.
- MSpline, RaisedCosineLog, RaisedCosineLinear returns an empty array, not an error.
- OrthExp raises another class specific error, which is that you need at least as many samples as basis function to define the basis (otherwise the ortonormalization would be not-defined).

Uniformity in behavior is hard to get given that different basis construction have different requirements.
My rationale here was try to surface the error and give a  basis-type specific explanation of the error which is understandable. 
If we try to get something that looks more uniform there are a few way:
- return an empty array for B-spline too instead of calling splev in case of empty input
- raise the error instead of returning an empty array for all of the basis function we have

Which one do you prefer?

Reply:
Gotcha. I'm inclined to work towards uniformity in behavior, and I don't have a strong preference here, but I think raising the error for all of them is the right way? because if someone calls `evaluate_on_grid(0)`, they've *really* misunderstood what it does. right?